packages/push-notifications/src/services/PushNotificationsService.ts

```diff
@@ -0,0 +1,31 @@
+import { ConnectionService } from '@aries-framework/core'
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { PushNotificationsGetDeviceInfoMessage, PushNotificationsSetDeviceInfoMessage } from '../messages'
+
+export interface DeviceInfo {
+  deviceToken: string
+  deviceVendor: DeviceVendor
+}
+
+export enum DeviceVendor {
+  Android = 'android',
+  Ios = 'ios',
+}
+
+@scoped(Lifecycle.ContainerScoped)
+export class PushNotificationsService {
+  public constructor(private connectionService: ConnectionService) {}
+
+  public async processSetDeviceInfo(deviceInfo: DeviceInfo, connectionId: string) {
+    const message = new PushNotificationsSetDeviceInfoMessage(deviceInfo)
+    const connection = await this.connectionService.getById(connectionId)
+    return { message, connection }
+  }
+
```

Comment:
The convention we have for services is 

`processXXX` for processing a received message and `createXXX` for creating a message to send. So in this case I'd use `createSetDeviceInfo`